led_strip/include/esp_private/led_strip_common.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <stdint.h>
+#include "esp_err.h"
+#include "led_strip_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief LED strip pixel order index
+ */
+typedef enum {
+    LED_PIXEL_INDEX_RED,     /*!< Red pixel index */
+    LED_PIXEL_INDEX_GREEN,   /*!< Green pixel index */
+    LED_PIXEL_INDEX_BLUE,    /*!< Blue pixel index */
+    LED_PIXEL_INDEX_WHITE,   /*!< White pixel index */
+    LED_PIXEL_INDEX_MAX      /*!< Max pixel index */
+} led_pixel_order_index_t;
+
+/**
+ * @brief Adjust LED color order
+ *
+ * @param led_pixel_offset Each pixel's offset
+ * @param config_pixel_order `config_pixel_order` parameter in LED strip configuration
+ * @param bytes_per_pixel bytes per pixel
+ * @return
+ *      - ESP_OK: Adjust LED color order successfully
+ *      - ESP_ERR_INVALID_ARG: Adjust LED color order failed because of invalid argument
+ */
+esp_err_t led_strip_adjust_color_order(uint8_t *led_pixel_offset, const uint8_t config_pixel_order, const uint8_t bytes_per_pixel);
```

Comment:
if you don't want to public this function (I guess so?), you can move the header file to under the `src` folder